core/src/main/java/com/graphhopper/util/shapes/Polygon.java

```diff
@@ -0,0 +1,193 @@
+package com.graphhopper.util.shapes;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Polygon implements Shape {
```

Comment:
We should avoid another Polygon class here. The problem here is probably that the current one is in the wrong package? com.graphhopper.routing.util.spatialrules

Or should we use the polygon from JTS?

Reply:
I implemented a new ```Polygon``` class because:
 - My class has the ```GROWN_FACTOR``` constant, which is used to increase bounds of ```Polygon```
 - The existing ```Polygon``` class doesn't implement ```Shape``` interface, so I cannot pass it to ```findEdgesInShape``` method.
 - The package com.graphhopper.util.shapes seem correct to me since it already has others geometric shapes.

I'm not sure if I can apply this changes to the existing ```Polygon``` class and use. If I can, please let me know and I do the changes.

I don't know if ```Polygon``` from JTS library has all features that I need, so I'm going to study the library if I can use it.

Reply:
> I'm going to study the library if I can use it.

This would be the preferred option

> If I can, please let me know and I do the changes.

Yes, if JTS is not possible please apply the changes to the existing class and e.g. use a default GROW_FACTOR of 1

Reply:
> The existing Polygon class doesn't implement Shape interface

I didn't implement the Shape interface because at that point we did not need all the methods supported by the shape interface like intersects etc. If using JTS does not work, feel free to extend the existing class :).

Reply:
So, I did the changes using JTS library and it works. The problem was the performance. Polygon class is almost 10x faster than JTS.

Maybe I did something wrong, I do not know. Anyway, I'm going to commit the changes using Polygon class.

Reply:
This is interesting thanks for letting us know. So which in-built methods did you use from JTS or where did the speed loss came from?

Reply:
btw: applying a (speed or access) change permanently via polgon would be also nice and we currently investigate to make this more streamlined in #1280

Reply:
@karussell I did not investigate where the performance problem is, just did a test with 484 requests. Using JTS the time was approximately 464 seconds versus 42 of the Polygon class

You can see the [changes using JTS solution](https://github.com/andreylh/graphhopper/commit/00157d0f7e08f0282676a879f1a2def65d63738d) in my forked repo

Reply:
Thanks! What is this method call: GeometryFactory.createPointFromInternalCoord ?

Reply:
It's to transform Coordinate into Geometry (Point) to be abble to pass into contains() method